applications/coreutils/tac.cpp

```diff
@@ -0,0 +1,122 @@
+#include <libsystem/cmdline/CMDLine.h>
+#include <libsystem/io/Path.h>
+#include <libsystem/io/Stream.h>
+#include <libutils/String.h>
+#include <libutils/StringBuilder.h>
+#include <libutils/Vector.h>
+
+static bool before = false;
+static char *separator = nullptr;
+
+static const char *usages[] = {
+    "NAME...",
+    "[OPTION]... NAME...",
+    nullptr,
+};
+
+static CommandLineOption options[] = {
+    COMMANDLINE_OPT_HELP,
+    COMMANDLINE_OPT_STRING("separator", 's', separator,
+                           "Choose the separator to be used instead of \\n",
+                           COMMANDLINE_NO_CALLBACK),
+    COMMANDLINE_OPT_BOOL("before", 'b', before,
+                         "Attach the separator before instead of after the string",
+                         COMMANDLINE_NO_CALLBACK),
+    COMMANDLINE_OPT_END};
+
+static CommandLine cmdline = CMDLINE(
+    usages,
+    options,
+    "Concatenate and print lines of file(s) in reverse.",
+    "If no filename provided read from input stream");
+
+Result tac(Stream *const input_stream)
+{
+    size_t read;
+    char buffer[1024];
+    char sep = (separator == nullptr ? '\n' : separator[0]);
```

Comment:
You should do that insead
```c++
if (!separator) {
    separator = "\n";
}
```

and then use directly the string instead of the `char`

Reply:
Im checking for a character to split so idk how to check for substring, would strstr be ok?

Reply:
Tac seperate on string and not on char 

```
       -s, --separator=STRING
              use STRING as the separator instead of newline
```
(from the man page of `tac`)

for your question, you should ask to @sleepy-monax

Reply:
Yes we don't have anything better than strstr right now ;)